src/cmd/services/m3coordinator/downsample/downsampler_test.go

```diff
@@ -127,6 +131,101 @@ func TestDownsamplerAggregationWithAutoMappingRulesFromNamespacesWatcher(t *test
 	testDownsamplerAggregation(t, testDownsampler)
 }
 
+func TestDownsamplerAggregationWithDownSampleAllDisable(t *testing.T) {
+	t.Parallel()
+
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	gaugeMetrics, _ := testGaugeMetrics(testGaugeMetricsOptions{})
+	require.Equal(t, 1, len(gaugeMetrics))
+
+	testDownsampler := newTestDownsampler(t, testDownsamplerOptions{
+		ingest: &testDownsamplerOptionsIngest{
+			gaugeMetrics: gaugeMetrics,
+		},
+		sampleAppenderOpts: &SampleAppenderOptions{
+			SeriesAttributes: ts.SeriesAttributes{M3Type: ts.M3MetricTypeCounter},
+			downsampleAll:    false,
```

Comment:
would you add samples are not used for downsampling and verify they are not down sampled?

Reply:
yes, above `gaugeMetrics: gaugeMetrics` is the ingest metric which is not down sampled:

```
expect: &testDownsamplerOptionsExpect{
			writes: []testExpectedWrite{},
		}
```